main_test.go

```diff
@@ -0,0 +1,147 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package section is responsible for splitting a program into its different
+// sections without attempting to parse the contents.s
+package main
+
+import (
+	"bytes"
+	"go/token"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewArgParser(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Success -d", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		arg := []string{"-d", "-p", "testdata/patch/error.patch", "testdata/test_files/lint_example/test2.go"}
+		_, err := argParser.ParseArgs(arg)
+		require.NoError(t, err)
+		patchFileName := arg[2]
+		assert.Equal(t, true, opts.Diff)
+		assert.Equal(t, []string{patchFileName}, opts.Patches)
+	})
+	t.Run("Success --diff", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		arg := []string{"--diff", "-p", "testdata/patch/error.patch", "testdata/test_files/lint_example/test2.go"}
+		patchFileName := arg[2]
+		_, err := argParser.ParseArgs(arg)
+		require.NoError(t, err)
+		assert.Equal(t, true, opts.Diff)
+		assert.Equal(t, []string{patchFileName}, opts.Patches)
+	})
+
+}
+
+func TestLoadPatches(t *testing.T) {
+	t.Parallel()
+	t.Run("Success", func(t *testing.T) {
+		opts := &options{Patches: []string{"testdata/patch/error.patch"},
+			Diff:           true,
+			DisplayVersion: false,
+			Verbose:        false,
+		}
+		opts.Args.Patterns = []string{"testdata/test_files/lint_example/test2.go"}
+		patch, err := loadPatches(token.NewFileSet(), opts, bytes.NewReader(nil))
+		assert.Equal(t, 1, len(patch))
+		require.NoError(t, err)
+
+		opts.Patches = []string{"testdata/patch/error.patch", "testdata/patch/time.patch"}
+		opts.Args.Patterns = []string{"testdata/test_files/lint_example/"}
+		require.NoError(t, err)
+		patch, err = loadPatches(token.NewFileSet(), opts, bytes.NewReader(nil))
+		require.NoError(t, err)
+		assert.Equal(t, 2, len(patch))
+	})
+	t.Run("Failure", func(t *testing.T) {
+		path := "testdata/patch/error1.patch"
+		opts := &options{Patches: []string{path},
+			Diff:           true,
+			DisplayVersion: false,
+			Verbose:        false,
+		}
+		opts.Args.Patterns = []string{"testdata/test_files/test2.go"}
+		_, err := loadPatches(token.NewFileSet(), opts, bytes.NewReader(nil))
+		assert.ErrorContains(t, err, path)
+		assert.ErrorContains(t, err, "no such file or directory")
+	})
+}
+
+func TestPreview(t *testing.T) {
+	t.Parallel()
+	t.Run("Success", func(t *testing.T) {
+		var tests = []struct {
```

Comment:
Don't need to use `var` form if the value is assigned right away.

```suggestion
		tests := []struct {
```